apps/web/src/components/AddContactModal/AddContactModal.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  Button,
+  FormControl,
+  FormErrorMessage,
+  FormLabel,
+  Heading,
+  Input,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+} from "@chakra-ui/react";
+import { useDynamicModalContext } from "@umami/components";
+import { type Contact } from "@umami/core";
+import { getNetworksForContracts } from "@umami/multisig";
+import {
+  contactsActions,
+  useAppDispatch,
+  useAsyncActionHandler,
+  useAvailableNetworks,
+  useValidateName,
+  useValidateNewContactPkh,
+} from "@umami/state";
+import { isValidContractPkh } from "@umami/tezos";
+import { type FC } from "react";
+import { useForm } from "react-hook-form";
+
+import { ModalCloseButton } from "../CloseButton";
+
+export const AddContactModal: FC<{
+  pkh?: string;
```

Comment:
I think we only use AddContactModal with preset pkh on web - so maybe we can make pkh required & remove some extra code & tests ?

(for new contacts we use button in the settings drawer)

<img width="633" alt="Screenshot 2024-11-08 at 14 37 41" src="https://github.com/user-attachments/assets/c1040ee0-a348-48d4-ae9b-08314321df26">